api/src/test/java/org/openmrs/api/VisitServiceTest.java

```diff
@@ -775,6 +763,33 @@ public void endVisit_shouldFailIfValidationErrorsAreFound() {
 		vs.endVisit(visit, cal.getTime());
 	}
 	
+	/**
+	 * @see VisitService#purgeVisit(Visit)
+	 */
+	@Test
+	public void purgeVisit_shouldEraseTheVisitFromTheDatabase() {
+		VisitService vs = Context.getVisitService();
+		Integer originalSize = vs.getVisits(null, null, null, null, null, null, null, null, null, true, true).size();
+		Visit visit = Context.getVisitService().getVisit(1);
+		
+		vs.purgeVisit(visit);
+		
+		assertEquals(originalSize - 1, vs.getVisits(null, null, null, null, null, null, null, null, null, true, true).size());
+	}
+	
+	/**
+	 * @see VisitService#purgeVisit(Visit)
+	 */
+	@Test
+	public void purgeVisit_shouldReturnWithoutVoidingIfVisitIsUnsaved() {
+		VisitService vs = Context.getVisitService();
+		Integer originalSize = vs.getVisits(null, null, null, null, null, null, null, null, null, true, true).size();
+		
+		vs.purgeVisit(new Visit());
+		
+		assertEquals((long)originalSize, vs.getVisits(null, null, null,null, null, null, null, null, null, true, true).size());
+	}
+	
 	/**
 	 * @see VisitService#purgeVisit(Visit)
 	 */
```

Comment:
the test `purgeVisit_shouldFailIfTheVisitHasEncountersAssociatedToIt` needs some changes, it doesnt test the branch correctly which you can see in the coverage. The test actually throws a `ValidationException` somewhere else, but since its an `APIException` the test passes although it doesnt test what we want 😉 

replace `e.setVisit(visit); Context.getEncounterService().saveEncounter(e);` with

```java
		visit.addEncounter(e);
		Context.getVisitService().saveVisit(visit);
```

and also add a newline before `purgeVisit` like you did in above tests.